src/main/java/org/threadly/litesockets/NoThreadSocketExecuter.java

```diff
@@ -52,25 +45,20 @@ protected void addThreadExecutorToClient(Client client) {
    * manually.
    */
   public void wakeup() {
-    if(isRunning()) {
-      commonSelector.wakeup();
-    }
-  }
-
-  @Override
-  public void flagNewWrite(Client client) {
-    updateClientOps(client);
+    checkRunning();
+    commonSelector.wakeup();
   }
 
   @Override
-  public void flagNewRead(Client client) {
-    updateClientOps(client);
-  }
-  
-  private void updateClientOps(Client client) {
+  public void setClientOperations(final Client client) {
     ArgumentVerifier.assertNotNull(client, "Client");
-    if(isRunning() && clients.containsKey(client.getChannel())) {
-      if(client.canWrite() && client.canRead()) {
+    if(isRunning() && !clients.containsKey(client.getChannel()) && !client.isClosed() && client.getClientsSocketExecuter() == this) {
```

Comment:
Ok, so fixing this lead me into a situation I forgot about.  Right now I allow manual construction of a Client/Server as long as you provide the SocketExecuter for it.  The issue is, there is no way to add them to the SocketExecuter maps if they are constructed this way.  

So we have 2 options

1. dont allow construction of client/servers manually, only through the SE
2. add a method to add the client/server to the SE if manually constructed.

the first options is kind of ideal to me, but I would want to hide the client/server constructors, but I am not sure how to do that and still let the SocketExecuters construct them as they are not in the same package.

The socket option is doable.  The client/server could be added pretty easy, assuming they have the same socketExecuter object set that they are being added too. But not sure it fits the design well.

Let me know what you think.